chain-signatures/node/src/protocol/message.rs

```diff
@@ -964,70 +937,33 @@ impl MessageOutbox {
-                }));
-            }
-        }
-
-        let mut errors = Vec::new();
-        let mut retry = VecDeque::new();
-        let mut uncompacted = 0;
-        let mut compacted = 0;
-        for task in send_tasks {
-            match task.await {
-                Ok(Ok(msgs_len)) => {
-                    uncompacted += msgs_len;
-                    compacted += 1;
-                }
-                Ok(Err((route, partition, err))) => {
-                    // since we failed, put back all the messages related to this
-                    retry.push_back((route, partition));
-                    errors.push(err);
-                }
-                Err(err) => {
-                    tracing::warn!(?err, "outbox: task failed to send message");
-                }
+                tokio::spawn(timeout(
+                    Duration::from_millis(cfg.message_timeout),
+                    async move {
+                        loop {
+                            let start = Instant::now();
+                            let Err(err) = client.msg(&url, &[&encrypted_partition]).await else {
+                                crate::metrics::SEND_ENCRYPTED_LATENCY
+                                    .with_label_values(&[account_id.as_str()])
+                                    .observe(start.elapsed().as_millis() as f64);
+                                tracing::debug!(elapsed = ?start.elapsed(), "outbox: sent encrypted messages");
+                                break;
+                            };
+
+                            crate::metrics::NUM_SEND_ENCRYPTED_FAILURE
+                                .with_label_values(&[account_id.as_str()])
+                                .inc_by(partition.messages.len() as f64);
+                            crate::metrics::FAILED_SEND_ENCRYPTED_LATENCY
+                                .with_label_values(&[account_id.as_str()])
+                                .observe(start.elapsed().as_millis() as f64);
+                            tracing::warn!(?err, "outbox: failed sending encrypted messages");
+
+                            tokio::time::sleep(Duration::from_millis(100)).await;
+                        }
+                    },
+                ));
             }
         }
-
-        if uncompacted > 0 {
-            tracing::debug!(
-                uncompacted,
-                compacted,
-                "sent messages in {:?}",
-                start.elapsed()
-            );
-        }
-
-        if !errors.is_empty() {
-            tracing::warn!(?errors, "outbox: failed sending encrypted messages");
-        }
-
-        // Add back the failed attempts for next time.
-        for (route, partition) in retry {
-            let entry = self.messages.entry(route).or_default();
-            entry.extend(
-                partition
-                    .messages
-                    .into_iter()
-                    .zip(partition.timestamps.into_iter()),
-            );
-        }
```

Comment:
nit: It looks like retrying later is no longer a thing, right? Would be nice to mention that in the commit message, if that's indeed the intention of this change.